RELEASE_PROCESS.md

```diff
@@ -0,0 +1,98 @@
+    - [Cutting a new release](#cutting-a-new-release)
+    - [Tagging Procedure](#tagging-procedure)
+  - [Patch release Procedure](#patch-release-procedure)
+
+This document outlines the release process for Babylon node (babylond)
+
+Babylon follows [semantic versioning](https://semver.org), but with the following deviations to account for state-machine and API breaking changes.
+
+- State-machine breaking changes & API breaking changes will result in an increase of the minor version Y (0.Y.z).
+- All other changes will result in an increase of the patch version Z (0.y.Z).
+
+## Breaking Changes
+
+A change is considered to be ***state-machine breaking*** if it requires a coordinated upgrade for the network to preserve state compatibility
+Note that when bumping the dependencies of [Cosmos SDK](https://github.com/cosmos/cosmos-sdk) and [IBC](https://github.com/cosmos/ibc-go),
+we will only treat patch releases as non state-machine breaking.
+
+A change is considered to be ***API breaking*** if it modifies the provided API. This includes events, queries, CLI interfaces.
+
+## Release Procedure
+
+A _release_ is an increment of the second number (eg: `v0.1.0` → `v0.2.0`)
+
+**Note**: Generally, PRs should target either `main` or a long-lived feature branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
+
+* Once the team feels that `main` is _**feature complete**_, we create a `release/v0.Y.x` branch (going forward known as release branch),
+  where `Y` is the minor version number, with patch part substituted to `x` (eg: v0.11.x).
+  * **PRs targeting directly a release branch can be merged _only_ when exceptional circumstances arise**.
+* We freeze the release branch from receiving any new features and focus on releasing a release candidate.
+  * Finish audits and reviews.
+  * Add more tests.
+  * Fix bugs as they are discovered.
+* After the team feels that the release branch works fine, we cut a release candidate.
+  * Create a new annotated git tag for a release candidate in the release branch (follow the [Tagging Procedure](#tagging-procedure)).
+  * The release verification on devnet must pass.
+  * When bugs are found, create a PR for `main`, and backport fixes to the release branch.
+  * Create new release candidate tags after bugs are fixed.
+* After the team feels the release candidate is ready, create a full release:
+  * **Note:** The final release MUST have the same commit hash as the latest corresponding release candidate.
+  * Create a new annotated git tag in the release branch (follow the [Tagging Procedure](#tagging-procedure))
+
+### Creating a new release branch
+
+- create a new release branch, e.g., `release/v0.10.x`
+    ```bash
+    git checkout main
+    git pull
+    git checkout -b release/v0.10.x
+    ```
+- push the release branch upstream
+    ```bash
+    git push
+    ```
+### Cutting a new release
+
+Before cutting a _**release candidate**_ (e.g., `v0.10.0-rc0`), the following steps are necessary:
```

Comment:
What about a normal release? When does a release candidate become a release?

Reply:
Here the answer is, when we feel confident about it. 

As described in other comments:
```
we tag a commit on release branch as v0.x.y-rc.0
we test/audit/ do what we want with this commit
if everything goes well we later tag this commit as v0.x.y
```